### PR TITLE
EES-3575 Add admin API for fetching legacy releases

### PR DESCRIPTION
**Pull request by benoutram**
This PR adds a new Admin API endpoint for fetching legacy releases for use on the 'Legacy releases' tab of the 'Manage publication' page which is part of the new Admin dashboard user journey.

We remove `LegacyReleases` from the Admin `PublicationViewModel` and `MyPublicationViewModel` to prevent fetching legacy releases unnecessarily where only a publication is required.

Until work on [EES-3217](https://dfedigital.atlassian.net/browse/EES-3217) is complete to update the admin dashboard user journey, the existing Manage Publication page which combines editing a publication with a view of the legacy releases needs to be maintained. This is done by making an additional call to get the legacy releases alongside the publication.

### UI test report

image

15 test failures in `Tests.Admin.Bau.Manage Users Page`.

### Other changes

* Renamed `PublicationLegacyReleaseEditPage.tsx` to `PublicationLegacyReleaseEditPage.test.tsx`
* Remove unused `PublicationId` from `LegacyReleaseViewModel` as I didn't feel it's necessary.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationRepositoryTests.cs [hunk 1]
```diff
@@ -1028,14 +1007,6 @@ await using (var context = InMemoryApplicationDbContext(contextId))
                 Assert.Single(viewModel.Methodologies);
                 Assert.Equal(methodologyVersion.Id, viewModel.Methodologies[0].Methodology.Id);
                 Assert.Equal(methodologyVersion.AlternativeTitle, viewModel.Methodologies[0].Methodology.Title);
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationRepositoryTests.cs:1009]: Going to have some conflicts here with Mark's PR which I've just finished reviewing :)

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs [hunk 2]
```diff
@@ -18,16 +20,24 @@ public LegacyReleaseController(ILegacyReleaseService legacyReleaseService)
         {
             _legacyReleaseService = legacyReleaseService;
         }
-        
-        [HttpGet("api/legacy-releases/{id}")]
+
+        [HttpGet("publications/{publicationId}/legacy-releases")]
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs:24]: Can use the `:guid` request parameter hint here to ensure it's the right type.

Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs:24]: Same for other new Controller methods

Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs]: Thanks, made this change.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs [hunk 3]
```diff
@@ -18,16 +20,24 @@ public LegacyReleaseController(ILegacyReleaseService legacyReleaseService)
         {
             _legacyReleaseService = legacyReleaseService;
         }
-        
-        [HttpGet("api/legacy-releases/{id}")]
+
+        [HttpGet("publications/{publicationId:guid}/legacy-releases")]
+        public async Task<ActionResult<List<LegacyReleaseViewModel>>> GetLegacyReleases(Guid publicationId)
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs:25]: Not a biggy, but could name this `ListLegacyReleases`. Same with the service method as well.

File: src/explore-education-statistics-admin/src/services/legacyReleaseService.ts [hunk 4]
```diff
@@ -22,6 +21,9 @@ const legacyReleaseService = {
   getLegacyRelease(id: string): Promise<LegacyRelease> {
     return client.get(`/legacy-releases/${id}`);
   },
+  getLegacyReleases(publicationId: string): Promise<LegacyRelease[]> {
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/services/legacyReleaseService.ts:24]: Similar to the backend - could name this `listLegacyReleases`

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs (hunk 3 above)
Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/LegacyReleaseController.cs]: Thanks, made this change now. 🙂

**Review by ntsim — APPROVED** [review]

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationRepositoryTests.cs (hunk 1 above)
Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationRepositoryTests.cs:1009]: Thanks for the heads up. I've just merged this branch into a temporary branch based on EES-3554 to see all the conflicts. There are in `PublicationServiceTests` and here in `PublicationRepositoryTests` but seem minor to resolve. Happy to pick up them up if EES-3554 is not merged in while @markhiveit is still here.

**Pull request merged**